app/src/main/java/com/example/filemanager/ListAdapter.kt

```diff
@@ -20,14 +29,16 @@ internal class ListAdapter : RecyclerView.Adapter<ListAdapter.ViewHolder>() {
     }
 
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        val item = items[position]
-        holder.name.text = item.fileName
+        val item = list[position]
+        holder.icon.setImageDrawable(item.icon?.let { ContextCompat.getDrawable(context, it) })
+        holder.fileName.text = item.fileName
+        holder.additionalInfo.text = item.additionalInfo
+        holder.itemView.setOnClickListener {
+            itemClickListener.onItemClickListener()
+        }
     }
 
-    fun replaceItems(items: List<ListModel>) {
-        this.items = items
-        notifyDataSetChanged()
+    override fun getItemCount(): Int {
+        return list.count()
```

Comment:
is `count()` the right method to get the size of a list? Shouldn't it be `size()` ?